tests/integration/pilot/vm_test.go

```diff
@@ -65,3 +76,113 @@ func TestVmOSPost(t *testing.T) {
 			}
 		})
 }
+
+func TestVMRegistrationLifecycle(t *testing.T) {
```

Comment:
This test runs for 30s which is tied for slowest pilot tests. Not sure how we can reduce it and keep the same coverage, but if we can it would be nice

Reply:
We can probably reduce the grace period value under test, as a start.